website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -137,6 +154,303 @@ Snowflake IO library supports following options that can be passed via the [comm
+  @Description("OAuth token. Required for OAuth authentication only.")
+  @Validation.Required(groups = AUTH_VALIDATION_GROUP)
+  String getOauthToken();
+
+  void setOauthToken(String oauthToken);
+
+  @Description("Password. Required for username/password authentication only.")
+  @Default.String("")
+  ValueProvider<String> getPassword();
+
+  void setPassword(ValueProvider<String> password);
+
+  @Description("Path to Private Key file. Required for Private Key authentication only.")
+  @Default.String("")
+  String getPrivateKeyPath();
+
+  void setPrivateKeyPath(String privateKeyPath);
+
+  @Description("Private key. Required for Private Key authentication only.")
+  @Default.String("")
+  ValueProvider<String> getRawPrivateKey();
+
+  void setRawPrivateKey(ValueProvider<String> rawPrivateKey);
+
+  @Description("Private Key's passphrase. Required for Private Key authentication only.")
+  @Default.String("")
+  ValueProvider<String> getPrivateKeyPassphrase();
+
+  void setPrivateKeyPassphrase(ValueProvider<String> keyPassphrase);
+
+  @Description("Warehouse to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getWarehouse();
+
+  void setWarehouse(ValueProvider<String> warehouse);
+
+  @Description("Database name to connect to. Optional.")
+  @Default.String("")
+  ValueProvider<String> getDatabase();
+
+  void setDatabase(ValueProvider<String> database);
+
+  @Description("Schema to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getSchema();
+
+  void setSchema(ValueProvider<String> schema);
+
+  @Description("Table to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getTable();
+
+  void setTable(ValueProvider<String> table);
+
+  @Description("Query to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getQuery();
+
+  void setQuery(ValueProvider<String> query);
+
+  @Description("Role to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getRole();
+
+  void setRole(ValueProvider<String> role);
+
+  @Description("Authenticator to use. Optional.")
+  @Default.String("")
+  String getAuthenticator();
+
+  void setAuthenticator(String authenticator);
+
+  @Description("Port number. Optional.")
+  @Default.String("")
+  String getPortNumber();
+
+  void setPortNumber(String portNumber);
+
+  @Description("Login timeout. Optional.")
+  @Default.String("")
+  String getLoginTimeout();
+
+  void setLoginTimeout(String loginTimeout);
+
+  @Description("Temporary GCS bucket name.")
+  ValueProvider<String> getStagingBucketName();
+
+  void setStagingBucketName(ValueProvider<String> stagingBucketName);
+
+  @Description("Storage integration name")
+  ValueProvider<String> getStorageIntegrationName();
+
+  void setStorageIntegrationName(ValueProvider<String> storageIntegrationName);
+
+  @Description("SnowPipe name. Optional.")
+  ValueProvider<String> getSnowPipe();
+
+  void setSnowPipe(ValueProvider<String> snowPipe);
+}
+{{< /highlight >}}
+### Using Pipeline options
```

Comment:
Also this section can be removed. Beam users will know how to use Pipeline options.

Reply:
I will do it

Reply:
I will remove also `Extending pipeline options`

Reply:
And `Running main command with Pipeline options`